build.gradle

```diff
@@ -24,19 +24,63 @@ plugins {
 
 allprojects {
     repositories {
-        google()
-        jcenter()
-
         maven {
             url "https://maven.mozilla.org/maven2"
+            content {
+                // We want to fetch all our known Mozilla groups from here.
+                includeGroup GroupMatching.GROUP_MOZILLA_APPSERVICES
+                includeGroup GroupMatching.GROUP_MOZILLA_COMPONENTS
+                includeGroup GroupMatching.GROUP_MOZILLA_GECKOVIEW
+            }
+        }
+
+        google() {
+            content {
+                // We want to fetch all AndroidX, Google and Firebase groups from here.
+                includeGroupByRegex GroupMatching.REGEX_ANDROIDX
+                includeGroupByRegex GroupMatching.REGEX_ANDROID
+                includeGroupByRegex GroupMatching.REGEX_GOOGLE_ANDROID
+                includeGroup GroupMatching.GROUP_GOOGLE_FIREBASE
+
+                // We never want to fetch Mozilla groups from here.
+                excludeGroupByRegex GroupMatching.REGEX_MOZILA
+            }
         }
 
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        // Servo
-        ////////////////////////////////////////////////////////////////////////////////////////////
+        mavenCentral() {
+            // We list Maven Central before jcenter since we prefer pulling artifacts from there if available. Maven
+            // Central has a more strict vetting process when adding new artifacts.
+
+            content {
+                // We never want to fetch Mozilla, AndroidX, Google or Firebase groups from Maven Central.
+                excludeGroupByRegex GroupMatching.REGEX_MOZILA
+                excludeGroupByRegex GroupMatching.REGEX_ANDROIDX
+                excludeGroupByRegex GroupMatching.REGEX_ANDROID
+                excludeGroup GroupMatching.GROUP_GOOGLE_FIREBASE
+
+                // We are not excluding REGEX_GOOGLE_ANDROID here since there's one group that is not getting published
+                // on maven.google.com so far (but on Maven Central, also jcenter but we prefer Maven Central):
+                // com.google.android.apps.common.testing.accessibility.framework
+                // See: https://github.com/google/Accessibility-Test-Framework-for-Android/issues/14
+            }
+        }
+
+        jcenter() {
+            content {
+                // We never want to fetch Mozilla, AndroidX, Google or Firebase groups from jcenter.
+                excludeGroupByRegex GroupMatching.REGEX_MOZILA
+                excludeGroupByRegex GroupMatching.REGEX_ANDROIDX
+                excludeGroupByRegex GroupMatching.REGEX_ANDROID
+                excludeGroupByRegex GroupMatching.REGEX_GOOGLE_ANDROID
+                excludeGroup GroupMatching.GROUP_GOOGLE_FIREBASE
+            }
+        }
 
         maven {
```

Comment:
I think the original reason why we moved this to the bottom was that we didn't want to hit `download.servo.org` early and for a lot of dependency requests. With the exclusion rules - does the order for *this* repository matter?

Reply:
With the current code, I don't think the order matters: we will only hit `download.servo.org` for the appropriate requests and those dependencies cannot be downloaded from elsewhere. However, moving it upward is defensive coding: if someone later changes the repositories above later such that SERVOVIEW is not excluded, we could be vulnerable to using the wrong dependency.

Note that if you move it up, you should add the exclusions too, though it's not strictly necessary (since `includeGroup...` means it'll only look at those specific dependencies).

Then again, this is getting sufficiently complicated that it'll be hard to change correctly later: perhaps just using `includeGroup`s without `excludeGroup`s but with an explicit comment like, "If this changes, be very careful because you could add a vulnerability!" or something is better than being explicit with our includeGroups and excludeGroups. @pocmo What do you think?